dataloader/dataloader.hpp

```diff
@@ -0,0 +1,245 @@
+   * @param augmentation Adds augmentation to training data only.
+   * @param augmentationProbability Probability of applying augmentation on dataset.
+   */
+  DataLoader(const std::string& dataset,
+             const bool shuffle,
+             const double ratio = 0.75,
+             const bool useScaler = true,
+             const std::vector<std::string> augmentation =
+                 std::vector<std::string>(),
+             const double augmentationProbability = 0.2);
+
+  /**
+   * Function to load and preprocess train or test data stored in CSV files.
+   * 
+   * @param datasetPath Path to the dataset.
+   * @param loadTrainData Boolean to determine whether data will be stored for
+   *                      training or testing. If true, data will be loaded for training.
+   *                      Note: This option augmentation to NULL, set ratio to 1 and
+   *                      scaler will be used to only transform the test data.
+   * @param shuffle Boolean to determine whether or not to shuffle the data.
+   * @param ratio Ratio for train-test split.
+   * @param useScaler Fits the scaler on training data and transforms dataset.
+   * @param dropHeader Drops the first row from CSV.
+   * @param startInputFeatures First Index which will be fed into the model as input.
+   *                           Note: Indicies are wrapped and -1 implies last
+   *                           column.
+   * @param endInputFeature Last Index which will be fed into the model as input.
+   *                        Note: Indicies are wrapped and -1 implies last
+   *                        column.
+   * @param startPredictionFeatures First Index which be predicted by the model as output.
+   *                                Note: Indicies are wrapped and -1 implies last
+   *                                column.
+   * @param endPredictionFeatures Last Index which be predicted by the model as output.
+   *                              Note: Indicies are wrapped and -1 implies last
+   *                              column.
+   * @param augmentation Vector strings of augmentations supported by mlpack.
+   * @param augmentationProbability Probability of applying augmentation to a particular cell.
+   */
+  void LoadCSV(const std::string &datasetPath,
+               const bool loadTrainData = true,
+               const bool shuffle = true,
+               const double ratio = 0.75,
+               const bool useScaler = false,
+               const bool dropHeader = false,
+               const int startInputFeatures = -1,
+               const int endInputFeatures = -1,
+               const int startPredictionFeatures = -1,
+               const int endPredictionFeatures = -1,
+               const std::vector<std::string> augmentation =
+                   std::vector<std::string>(),
+               const double augmentationProbability = 0.2);
+
+  //! Get the Training Dataset.
+  DatasetX TrainX() const { return trainX; }
```

Comment:
I'm wondering if we should use a more descriptive name, like `TrainSet` or `TrainPredictors` and `TrainLabels` or `TrainResponses`, let me know what you think.

Reply:
I guess we could also return a tuple.

Reply:
Yes, Maybe `TrainFeatures` ? and `TrainLabels` make sense.

> I guess we could also return a tuple.

Yes, that could be added.
Thanks for the review.
Regards.

Reply:
I like the naming.